File: carddemo-batch-python/src/carddemo_batch/models/account.py
"""Account model - migrated from COBOL copybook CVACT01Y.cpy (300 bytes)."""

from decimal import Decimal

from sqlalchemy import Numeric, String
from sqlalchemy.orm import Mapped, mapped_column

from carddemo_batch.models.base import Base


class Account(Base):
    """
    Account entity representing the ACCTDATA VSAM file.
    
    Migrated from COBOL copybook CVACT01Y.cpy.
    Record length: 300 bytes
    Primary key: ACCT-ID (11 digits)
    """
    __tablename__ = "acctdata"

    acct_id: Mapped[int] = mapped_column(primary_key=True)
    acct_active_status: Mapped[str] = mapped_column(String(1), nullable=True)
    acct_curr_bal: Mapped[Decimal] = mapped_column(Numeric(12, 2), nullable=True)
    acct_credit_limit: Mapped[Decimal] = mapped_column(Numeric(12, 2), nullable=True)
    acct_cash_credit_limit: Mapped[Decimal] = mapped_column(Numeric(12, 2), nullable=True)
    acct_open_date: Mapped[str] = mapped_column(String(10), nullable=True)
    acct_expiraion_date: Mapped[str] = mapped_column(String(10), nullable=True)
    acct_reissue_date: Mapped[str] = mapped_column(String(10), nullable=True)
    acct_curr_cyc_credit: Mapped[Decimal] = mapped_column(Numeric(12, 2), nullable=True)
    acct_curr_cyc_debit: Mapped[Decimal] = mapped_column(Numeric(12, 2), nullable=True)
    acct_addr_zip: Mapped[str] = mapped_column(String(10), nullable=True)
    acct_group_id: Mapped[str] = mapped_column(String(10), nullable=True)

    def __repr__(self) -> str:
        return f"Account(acct_id={self.acct_id}, status={self.acct_active_status})"

File: app-java/src/main/java/com/carddemo/model/Account.java
package com.carddemo.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

/**
 * JPA entity mapped from COBOL copybook CVACT01Y.cpy (ACCOUNT-RECORD, RECLN 300).
 */
@Entity
@Table(name = "accounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Account {

    /** ACCT-ID PIC 9(11) */
    @Id
    @Column(name = "acct_id")
    private Long acctId;

    /** ACCT-ACTIVE-STATUS PIC X(01) */
    @Column(name = "acct_active_status", length = 1)
    private String acctActiveStatus;

    /** ACCT-CURR-BAL PIC S9(10)V99 */
    @Column(name = "acct_curr_bal", precision = 12, scale = 2)
    private BigDecimal acctCurrBal;

    /** ACCT-CREDIT-LIMIT PIC S9(10)V99 */
    @Column(name = "acct_credit_limit", precision = 12, scale = 2)
    private BigDecimal acctCreditLimit;

    /** ACCT-CASH-CREDIT-LIMIT PIC S9(10)V99 */
    @Column(name = "acct_cash_credit_limit", precision = 12, scale = 2)
    private BigDecimal acctCashCreditLimit;

    /** ACCT-OPEN-DATE PIC X(10) */
    @Column(name = "acct_open_date", length = 10)
    private String acctOpenDate;

    /** ACCT-EXPIRAION-DATE PIC X(10) — typo preserved from COBOL */
    @Column(name = "acct_expiration_date", length = 10)
    private String acctExpirationDate;

    /** ACCT-REISSUE-DATE PIC X(10) */
    @Column(name = "acct_reissue_date", length = 10)
    private String acctReissueDate;

    /** ACCT-CURR-CYC-CREDIT PIC S9(10)V99 */
    @Column(name = "acct_curr_cyc_credit", precision = 12, scale = 2)
    private BigDecimal acctCurrCycCredit;

    /** ACCT-CURR-CYC-DEBIT PIC S9(10)V99 */
    @Column(name = "acct_curr_cyc_debit", precision = 12, scale = 2)
    private BigDecimal acctCurrCycDebit;

    /** ACCT-ADDR-ZIP PIC X(10) */
    @Column(name = "acct_addr_zip", length = 10)
    private String acctAddrZip;

    /** ACCT-GROUP-ID PIC X(10) */
    @Column(name = "acct_group_id", length = 10)
    private String acctGroupId;

    /** Optimistic-lock version — replaces CICS READ UPDATE / REWRITE */
    @Version
    private Long version;
}
